### ISSUE TITLE
When maximizing it pushes the window to the next workspace

### ISSUE THREAD

**Issue by lambduli** [rliang/gnome-shell-extension-maximize-to-workspace#12]
Hello, thanks for much needed extension. I really like it.
Though I encountered this behaviour, which I don't find productive:

### Expected behavior

When I have a workspace with one maximized window and then I open new window and maximize it (or it opens maximized). Extension should create new workspace right after this workspace and push the new window on it - in this case I should be moved to this new workspace.
(Or push the old window to the new one - then I can stay on this workspace.)

### Actual behavior

When I have a workspace with one window maximized and then I open another application and maximize it (or it opens maximized) my current maximized window is pushed to the next workspace - which is already occupied by something else most of the time.

Which leads to having more than one maximized window on the next workspace. Or generally having the maximized window and whatever was on this workspace originaly before the pushing took place.

### Steps to reproduce the behavior

- Open window on workspace
- maximize it
- switch to the next workspace and open some window there (doesn't matter if maximized or not)
- get back to your maximized (first) workspace and open new window -> maximize it
- inspect where your windows are -> new window is on your current (first) workspace and old maximized window is cramped with the other application on the second workspace

### (Recommended) GNOME version

3.36.5

**Comment by lambduli** [comment]
Hello again, I have implemented it myself and would gladly submit PR to you.

Only thing is - version of the extension I've downloaded from the extension.gnome.org seems to be different than the published one here.
That is - judging by the content of the `extension.js` file.

If you update the repo or give me some instructions which version do you prefer I can fork this project and submit a PR with the fix.

Behaviour after my change:

When a window is maximized it is put to the solo workspace on the right of the original workspace.

Implementaiton:

When a window is maximized, new workspace is created and moved BEFORE the current workspace, then all the other windows are moved to the left instead of right - which leaves them in the new empty workspace on the left of the currently maximized window.
This seems like the behaviour which makes the most sense and does not merge existing workspaces with moving windows.

**Comment by lambduli** [comment]
I've now just realized it actually already has been fixed by one of the previous PRs. I am closing the issue as it's not really needed. I am also gonna install the extensions from the github because published version does not have this feature.

Edit: Unfortunately the last version puts the maximized version at the end of the workspace stack which is not something I prefer. So I am probably gonna fork this extension and implement behaviour which makes sense for me.
Thanks anyway - at least I don't have to start from the scratch.
